husky_musher/blueprints/app.py

```diff
@@ -66,20 +66,23 @@ def render_redirect(
         # them to the next, uncompleted survey in the queue.
         event = "enrollment_arm_1"
         instrument = "enrollment_questions"
-
+        survey_link = client.generate_survey_link(
+            redcap_record["record_id"], event, instrument
+        )
```

Comment:
It looks like `generate_survey_link` is being called even if a user has not completed their registration; then, a second link is generated if the user _has_ completed their registration. Is that the desired behavior?

Does `generate_survey_link` redirect the user to a form to register?

Reply:
Never mind, I see that I am presented with "Enrollment Questions" the first time I visit this survey.

Reply:
Good catches - what I did isn't ideal and generates an unnecessary API call. 

The way it flows now is that the value of `survey_link` is initially set to be the URL of the enrollment survey for a user. 

If that user has already completed the enrollment form, then `survey_link` is reset to be the link to the surveyQueue for that user.

Any suggestions on how to better restructure?

Reply:
Hmm, I'll give it some thought. In the meantime I'll merge this as-is.